internal/predicate/nonadminbackupstoragelocationrequest_predicate.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package predicate
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/event"
+
+	"github.com/migtools/oadp-non-admin/internal/common/function"
+)
+
+const nonAdminBackupStorageLocationRequestPredicateKey = "NonAdminBackupStorageLocationRequestPredicate"
+
+// NonAdminBackupStorageLocationRequestPredicate contains event filters for Non Admin Backup Storage Location Request objects
+type NonAdminBackupStorageLocationRequestPredicate struct{}
+
+// Create event filter accepts all NonAdminBackupStorageLocationRequest create events
+func (NonAdminBackupStorageLocationRequestPredicate) Create(ctx context.Context, evt event.CreateEvent) bool {
```

Comment:
can be deleted? only Update and Delete are used, right?

Reply:
The delete causes the controller to re-create, so it is handled. This is to ensure we always have corresponding Request, including situations where controller is started later.

If admin Deletes the `NonAdminBackupStorageLocationRequest` then it has lost information about previous approval and it reverts back to `pending`, deleting corresponding BSL objects.